accounts/static/accounts/js/profile.js

```diff
@@ -0,0 +1,354 @@
+    overlay.classList.add("hidden");
+    remove_friend_modal.style.display = "none";
+}
+/**
+ * This function sends a POST request to remove a friend from the users friends list.
+ * @param {String} friend_username 
+ */
+async function removeFriend(friend_username){
+    // Make a POST request to remove friend from friends list
+    const csrfToken = await getCSRFToken();
+    fetch("/accounts/remove/friend/", {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+            'X-CSRFToken': csrfToken
+        },
+        body: JSON.stringify({ username: friend_username })
+    })
+    .then(response => {
+        if(response.status == 200){
+            // Hide buttons
+            remove_friend_button.style.display = "none";
+            remove_reject_button.style.display = "none";
+            remove_friend_response.style.color = "green";
+        }else{
+            remove_friend_response.style.color = "red";
+        }
+        return response.text();
+    })
+    .then(data => {
+        // Display response message
+        remove_friend_response.innerHTML = data;
+    })
+    .catch(error => {
+        // Display error message
+        console.log("An error occured during the remove friend request: " + error);
+    });
+}
+
+
+/**
+ * 
+ * Send friend request functionality
+ * 
+ */
+
+// elements
+const add_friend_modal = document.getElementById("add_friend_modal");
+const add_friend_response = document.getElementById("add_friend_response");
+const send_friend_username = document.getElementById("send_friend_username");
+
+/**
+ * This function opens/resets the modal for sending a friend request
+ * to another user and blurs the background.
+ */
+function openAddFriendModal(){
+    // Make response/username input empty
+    add_friend_response.innerHTML = "";
+    send_friend_username.value = "";
+    // Make modal visible, blur background
+    add_friend_modal.style.display = "block";
+    overlay.classList.remove("hidden");
+}
+/**
+ * This function closes the modal for sending a friend request.
+ */
+function closeAddFriendModal(){
+    overlay.classList.add("hidden");
+    add_friend_modal.style.display = "none";
+}
+/**
+ * This function sends a POST request to send a friend request to another user.
+ * @param {String} username the username of the user to send a friend request to
+ */
+async function sendFriendRequest(username){
+    // Make a POST request to send friend request
+    const csrfToken = await getCSRFToken();
+    const name_input = send_friend_username.value;
+    console.log("Sending friend request to: " + name_input)
+    fetch("/accounts/send/friend/", {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+            'X-CSRFToken': csrfToken
+        },
+        body: JSON.stringify({ username: name_input })
+    })
+    .then(response => {
+        if(response.status == 200){
```

Comment:
set to green but if response.status isn't 200, change to red?

Reply:
Yeah it's just another visual indicator (besides the text content) that the result either went good or bad. My view always has text responses so the user knows what happened with their response and/or why their response was good or bad